### PR TITLE
Vars and optional route table

### PR DESCRIPTION
**Pull request by geekbass**
- Modified the vars to be more in tune with our strategy
- Updated the vars and the spreadsheet so we are good to go with CI
- Made the route_table_id for both local and remote VPC optional since we are able to get this from data sources. Users choice.
- Move data source aws_route_tables to aws_route_table as we are not using or are in need of lists of attributes but specific attributes of a specific route table. See commit 067a9c06596b57fc2b0f676c0a1d660918ce09cc for more info on thought process.

### CODE REVIEW

**Review by geekbass — COMMENTED** [review]
This was an issue within my own main.tf. ooof! Ignore the comments. This working as expected.

File: main.tf [hunk 1]
```diff
@@ -64,81 +64,72 @@
...
-  provider    = "aws.this"
-  type        = "ingress"
-  from_port   = 0
-  to_port     = 65535
-  protocol    = "all"
-  cidr_blocks = ["${var.peer_cidr_block}"]
-
-  security_group_id = "${var.this_security_group_id}"
-}
-
```
Comment (bernadinm) [inline main.tf:134]: Hey @fatz @geekbass, Will this removal still give us the single terraform apply ability?

File: README.md [hunk 2]
```diff
@@ -15,18 +15,18 @@ module "vpc_single_region_peering" {
...
-  peer_vpc_id              = "vpc-bbbbbbbb"
-  peer_cidr_block          = "10.0.0.0/16"
-  peer_main_route_table_id = "rtb-aaaaaaaa"
-  peer_security_group_id   = "sg-11111111"
-  this_cidr_block          = "10.1.0.0/16"
-  this_main_route_table_id = "rtb-bbbbbbbb"
-  this_security_group_id   = "sg-00000000"
-  this_vpc_id              = "vpc-aaaaaaaa"
+  remote_vpc_id              = "vpc-bbbbbbbb"
+  remote_subnet_range          = "10.0.0.0/16"
```
Comment (bernadinm) [inline README.md]: minor formatting

File: README.md [hunk 3]
```diff
@@ -15,18 +15,18 @@ module "vpc_single_region_peering" {
...
+  remote_main_route_table_id = "rtb-aaaaaaaa"
+  remote_security_group_id   = "sg-11111111"
+  local_subnet_range          = "10.1.0.0/16"
```
Comment (bernadinm) [inline README.md]: minor formatting

File: README.md [hunk 4]
```diff
@@ -41,18 +41,18 @@ module "vpc_cross_region_peering" {
...
-  peer_vpc_id              = "vpc-bbbbbbbb"
-  peer_cidr_block          = "10.0.0.0/16"
-  peer_main_route_table_id = "rtb-aaaaaaaa"
-  peer_security_group_id   = "sg-11111111"
-  this_cidr_block          = "10.1.0.0/16"
-  this_main_route_table_id = "rtb-bbbbbbbb"
-  this_security_group_id   = "sg-00000000"
-  this_vpc_id              = "vpc-aaaaaaaa"
+  remote_vpc_id              = "vpc-bbbbbbbb"
+  remote_subnet_range          = "10.0.0.0/16"
```
Comment (bernadinm) [inline README.md]: minor formatting

File: README.md [hunk 5]
```diff
@@ -41,18 +41,18 @@ module "vpc_cross_region_peering" {
...
+  remote_main_route_table_id = "rtb-aaaaaaaa"
+  remote_security_group_id   = "sg-11111111"
+  local_subnet_range          = "10.1.0.0/16"
```
Comment (bernadinm) [inline README.md]: minor formatting

File: main.tf [hunk 6]
```diff
@@ -16,18 +16,18 @@
...
- *   peer_vpc_id              = "vpc-bbbbbbbb"
- *   peer_cidr_block          = "10.0.0.0/16"
- *   peer_main_route_table_id = "rtb-aaaaaaaa"
- *   peer_security_group_id   = "sg-11111111"
- *   this_cidr_block          = "10.1.0.0/16"
- *   this_main_route_table_id = "rtb-bbbbbbbb"
- *   this_security_group_id   = "sg-00000000"
- *   this_vpc_id              = "vpc-aaaaaaaa"
+ *   remote_vpc_id              = "vpc-bbbbbbbb"
+ *   remote_subnet_range          = "10.0.0.0/16"
```
Comment (bernadinm) [inline main.tf]: minor formatting

File: main.tf [hunk 7]
```diff
@@ -16,18 +16,18 @@
...
+ *   remote_main_route_table_id = "rtb-aaaaaaaa"
+ *   remote_security_group_id   = "sg-11111111"
+ *   local_subnet_range          = "10.1.0.0/16"
```
Comment (bernadinm) [inline main.tf]: minor formatting

File: main.tf [hunk 8]
```diff
@@ -42,18 +42,18 @@
...
- *   peer_vpc_id              = "vpc-bbbbbbbb"
- *   peer_cidr_block          = "10.0.0.0/16"
- *   peer_main_route_table_id = "rtb-aaaaaaaa"
- *   peer_security_group_id   = "sg-11111111"
- *   this_cidr_block          = "10.1.0.0/16"
- *   this_main_route_table_id = "rtb-bbbbbbbb"
- *   this_security_group_id   = "sg-00000000"
- *   this_vpc_id              = "vpc-aaaaaaaa"
+ *   remote_vpc_id              = "vpc-bbbbbbbb"
+ *   remote_subnet_range          = "10.0.0.0/16"
```
Comment (bernadinm) [inline main.tf]: minor formatting

File: main.tf [hunk 9]
```diff
@@ -42,18 +42,18 @@
  *   source = "dcos-terraform/vpc-peering/aws"
  * 
  *   providers = {
- *     aws.this = "aws.src"
- *     aws.peer = "aws.dst"
+ *     aws.local = "aws.src"
```
Comment (bernadinm) [inline main.tf]: minor formatting

File: main.tf [hunk 10]
```diff
@@ -16,18 +16,18 @@
  *   source = "dcos-terraform/vpc-peering/aws"
  * 
  *   providers = {
- *     aws.this = "aws"
- *     aws.peer = "aws"
+ *     aws.local = "aws"
```
Comment (bernadinm) [inline main.tf:19]: minor formatting

File: README.md [hunk 11]
```diff
@@ -41,18 +41,18 @@ module "vpc_cross_region_peering" {
   source = "dcos-terraform/vpc-peering/aws"
 
   providers = {
-    aws.this = "aws.src"
-    aws.peer = "aws.dst"
+    aws.local = "aws.src"
```
Comment (bernadinm) [inline README.md]: minor formatting

File: README.md [hunk 12]
```diff
@@ -15,18 +15,18 @@ module "vpc_single_region_peering" {
   source = "dcos-terraform/vpc-peering/aws"
 
   providers = {
-    aws.this = "aws"
-    aws.peer = "aws"
+    aws.local = "aws"
```
Comment (bernadinm) [inline README.md:18]: minor formatting

**Review by bernadinm — CHANGES_REQUESTED** [review]
perhaps we can run the `terraform fmt` and `docker run -ti -v$(pwd):/a --entrypoint /usr/bin/terraform-docs fatz/terraform-jenkins md /a | tr -d '\r' > README.md`

File: main.tf (hunk 1 above)
Comment (geekbass) [reply main.tf:134]: Yes!

**Review by geekbass — COMMENTED** [review]
thanks for the attention to detail here @bernadinm !

**Review by fatz — APPROVED** [review]
lgtm

**Review by bernadinm — APPROVED** [review]
LGTM!

**Pull request merged**